legend-pure-code-compiled-core/src/main/resources/core/pure/persistence/metamodel.pure

```diff
@@ -0,0 +1,409 @@
+meta::pure::persist::metamodel::batch::milestoned::validity::OpaqueValidityMilestoningScheme extends ValidityMilestoningScheme
+{
+}
+
+// validity provider
+
+Class
+<<typemodifiers.abstract>>
+{doc.doc = 'Strategy for representing the valid (business) temporal dimension in the underlying system.'}
+meta::pure::persist::metamodel::batch::milestoned::validity::derivation::ValidityDerivation
+{
+}
+
+Class
+{doc.doc = 'Source data provide valid from datetime.'}
+meta::pure::persist::metamodel::batch::milestoned::validity::derivation::SourceSpecifiesValidFromDate extends ValidityDerivation
+{
+   sourceValidDateTimeFromProperty: Property<Nil, DateTime|1>[1];
+}
+
+Class
+{doc.doc = 'Source data provide valid from and thru datetimes.'}
+meta::pure::persist::metamodel::batch::milestoned::validity::derivation::SourceSpecifiesValidFromAndThruDate extends ValidityDerivation
+{
+   sourceValidDateTimeFromProperty: Property<Nil, Any|1>[1];
+   sourceValidDateTimeThruProperty: Property<Nil, Any|1>[1];
+}
+
+/**********
+ * batch mode: milestoned - snapshot
+ **********/
+
+Class
+<<typemodifiers.abstract>>
+{doc.doc = 'Values in the incoming batch represent the full dataset and will replace the existing dataset non-destructively. New incoming values represent a logical create; updated incoming values represent a logical update; missing incoming values represent a logical delete in the transaction (processing) dimension.' }
+meta::pure::persist::metamodel::batch::milestoned::snapshot::SnapshotMilestoned extends Milestoned
+{
+}
+
+Class
+{doc.doc = 'Values in the incoming batch represent the full dataset and will replace the existing dataset non-destructively. The data have only a transaction (processing) dimension and no validity (business) dimension.'}
+meta::pure::persist::metamodel::batch::milestoned::snapshot::UnitemporalSnapshotMilestoned extends SnapshotMilestoned
+{
+}
+
+Class
+{doc.doc = 'Values in the incoming batch represent the full dataset and will replace the existing dataset non-destructively. The data have a transaction (processing) dimension and a validity (business) dimension.'}
+meta::pure::persist::metamodel::batch::milestoned::snapshot::BitemporalSnapshotMilestoned extends SnapshotMilestoned
+{
+   validityMilestoningScheme: ValidityMilestoningScheme[1];
+   validityDerivation: ValidityDerivation[1];
+}
+
+/**********
+ * batch mode: milestoned - delta
+ **********/
+
+Class
+<<typemodifiers.abstract>>
+{doc.doc = 'Values in the incoming batch represent a delta to be applied on top of the existing dataset non-destructively. New incoming values represent a logical create; updated incoming values represent a logical update; missing incoming values represent a no-op in the transaction (processing) dimension. Additional merge semantics, including logical deletes, can be supported via a merge scheme.' }
+meta::pure::persist::metamodel::batch::milestoned::delta::DeltaMilestoned extends Milestoned
+{
```

Comment:
Is this missing the field to define the MergeScheme?  Otherwise, the model does not seem to leverage MergeScheme classes.

Reply:
Good catch -- it's added in the latest commit.